llvm/lib/Transforms/IPO/SampleProfileMatcher.cpp

```diff
@@ -189,7 +238,9 @@ LocToLocMap SampleProfileMatcher::longestCommonSequence(
         X = V[Index(K - 1)] + 1;
       Y = X - K;
       while (X < Size1 && Y < Size2 &&
-             AnchorList1[X].second == AnchorList2[Y].second)
+             functionMatchesProfile(AnchorList1[X].second,
+                                    AnchorList2[Y].second,
+                                    !MatchUnusedFunction))
```

Comment:
The negation is slightly confusing. Perhaps add a comment : !MatchUnusedFunction /* matched functions only */